src/main/java/io/cdap/plugin/gcp/spanner/source/SpannerRecordReader.java

```diff
@@ -101,5 +102,9 @@ public void close() throws IOException {
     LOG.trace("Closing Record reader");
     bytesRead.increment(counter.getValue());
     resultSet.close();
+    if (spanner != null) {
+      spanner.close();
```

Comment:
were these messages due to spanner resource not being closed?

Reply:
Exactly. These error logs were caused because of spanner resources not being closed properly.

Reply:
Got it, thanks!